From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: u9g <git@u9g.dev>
Date: Fri, 15 Jul 2022 22:25:17 -0400
Subject: [PATCH] Add UnloadWorldResult


diff --git a/src/main/java/dev/u9g/FunctionResult.java b/src/main/java/dev/u9g/FunctionResult.java
new file mode 100644
index 0000000000000000000000000000000000000000..f9970b8ddcc6dac057d0ae1a042c85f491d3fba2
--- /dev/null
+++ b/src/main/java/dev/u9g/FunctionResult.java
@@ -0,0 +1,5 @@
+package dev.u9g;
+
+public interface FunctionResult {
+    boolean isSuccessful();
+}
diff --git a/src/main/java/dev/u9g/UnloadWorldResult.java b/src/main/java/dev/u9g/UnloadWorldResult.java
new file mode 100644
index 0000000000000000000000000000000000000000..7c6abe665b96f0ed772fd30e1f0ab777eedfacc5
--- /dev/null
+++ b/src/main/java/dev/u9g/UnloadWorldResult.java
@@ -0,0 +1,23 @@
+package dev.u9g;
+
+public enum UnloadWorldResult implements FunctionResult {
+    WORLD_IS_NULL,
+    WORLD_IS_NOT_LOADED,
+    IS_MAIN_WORLD,
+    WORLD_HAS_PLAYERS,
+    IS_CANCELLED,
+    SUCCESS(true);
+
+    private boolean isSuccessful = false;
+
+    UnloadWorldResult() {}
+
+    UnloadWorldResult(boolean isSuccessful) {
+        this.isSuccessful = isSuccessful;
+    }
+
+    @Override
+    public boolean isSuccessful() {
+        return isSuccessful;
+    }
+}
diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index a87399fa4838d4b2c1ff9cc35d433ae76cc149bf..e9c1a2ec2e4510f1efc48709657501dd4f2ff426 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -774,6 +774,10 @@ public final class Bukkit {
      * @param save whether to save the chunks before unloading
      * @return true if successful, false otherwise
      */
+    // Sliver start - UnloadWorldResult
+    public static dev.u9g.UnloadWorldResult sliver_unloadWorld(@NotNull String name, boolean save) { return server.sliver_unloadWorld(name, save); } // Sliver - UnloadWorldResult
+    @Deprecated
+    // Sliver end
     public static boolean unloadWorld(@NotNull String name, boolean save) {
         return server.unloadWorld(name, save);
     }
@@ -785,6 +789,10 @@ public final class Bukkit {
      * @param save whether to save the chunks before unloading
      * @return true if successful, false otherwise
      */
+    // Sliver start - UnloadWorldResult
+    public static dev.u9g.UnloadWorldResult sliver_unloadWorld(@NotNull World world, boolean save) { return server.sliver_unloadWorld(world, save); } // Sliver - UnloadWorldResult
+    @Deprecated
+    // Sliver end
     public static boolean unloadWorld(@NotNull World world, boolean save) {
         return server.unloadWorld(world, save);
     }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 79b26045a68ebb9b01e5bd06abbccaaef5489777..402c461c10957534459b96b0194b9e41bf21c4c4 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -641,6 +641,10 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      * @param save whether to save the chunks before unloading
      * @return true if successful, false otherwise
      */
+    // Sliver start - UnloadWorldResult
+    public dev.u9g.UnloadWorldResult sliver_unloadWorld(@NotNull String name, boolean save);
+    @Deprecated
+    // Sliver end
     public boolean unloadWorld(@NotNull String name, boolean save);
 
     /**
@@ -650,6 +654,10 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      * @param save whether to save the chunks before unloading
      * @return true if successful, false otherwise
      */
+    // Sliver start - UnloadWorldResult
+    public dev.u9g.UnloadWorldResult sliver_unloadWorld(@NotNull World world, boolean save);
+    @Deprecated
+    // Sliver end
     public boolean unloadWorld(@NotNull World world, boolean save);
 
     /**
