From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: u9g <git@u9g.dev>
Date: Fri, 15 Jul 2022 22:26:08 -0400
Subject: [PATCH] Add UnloadWorldResult


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 8a248e34fce1e04de4460f98b7627df495c66af1..3f6352716fcc113d60b8ebe82c51cf898ed324b6 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1271,6 +1271,7 @@ public final class CraftServer implements Server {
         return internal.getWorld();
     }
 
+    @Override public dev.u9g.UnloadWorldResult sliver_unloadWorld(String name, boolean save) { return sliver_unloadWorld(this.getWorld(name), save); } // Sliver - UnloadWorldResult
     @Override
     public boolean unloadWorld(String name, boolean save) {
         return this.unloadWorld(this.getWorld(name), save);
@@ -1278,29 +1279,36 @@ public final class CraftServer implements Server {
 
     @Override
     public boolean unloadWorld(World world, boolean save) {
+        // Sliver start - UnloadWorldResult
+        return sliver_unloadWorld(world, save).isSuccessful();
+    }
+
+    @Override
+    public dev.u9g.UnloadWorldResult sliver_unloadWorld(World world, boolean save) {
+        // Sliver end
         if (world == null) {
-            return false;
+            return dev.u9g.UnloadWorldResult.WORLD_IS_NULL; // Sliver - UnloadWorldResult
         }
 
         ServerLevel handle = ((CraftWorld) world).getHandle();
 
         if (!(console.levels.containsKey(handle.dimension()))) {
-            return false;
+            return dev.u9g.UnloadWorldResult.WORLD_IS_NOT_LOADED; // Sliver - UnloadWorldResult
         }
 
         if (handle.dimension() == net.minecraft.world.level.Level.OVERWORLD) {
-            return false;
+            return dev.u9g.UnloadWorldResult.IS_MAIN_WORLD; // Sliver - UnloadWorldResult
         }
 
         if (handle.players().size() > 0) {
-            return false;
+            return dev.u9g.UnloadWorldResult.WORLD_HAS_PLAYERS; // Sliver - UnloadWorldResult
         }
 
         WorldUnloadEvent e = new WorldUnloadEvent(handle.getWorld());
         this.pluginManager.callEvent(e);
 
         if (e.isCancelled()) {
-            return false;
+            return dev.u9g.UnloadWorldResult.IS_CANCELLED; // Sliver - UnloadWorldResult
         }
 
         try {
@@ -1317,7 +1325,7 @@ public final class CraftServer implements Server {
 
         this.worlds.remove(world.getName().toLowerCase(java.util.Locale.ENGLISH));
         console.levels.remove(handle.dimension());
-        return true;
+        return dev.u9g.UnloadWorldResult.SUCCESS; // Sliver - UnloadWorldResult
     }
 
     public DedicatedServer getServer() {
