From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: u9g <git@u9g.dev>
Date: Sun, 17 Jul 2022 03:04:44 -0400
Subject: [PATCH] Add Player#getFishHook


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
index 83c5314ed3a3f09162bca47902c4e749fab940a0..3f506824ae8d58b3fae839d5e65d0752d92846c4 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
@@ -778,4 +778,5 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
     public void setStarvationRate(int i) {
         this.getHandle().getFoodData().starvationRate = i;
     }
+    @Override public org.bukkit.entity.FishHook getFishHook() { return getHandle().fishing != null ? (org.bukkit.entity.FishHook) getHandle().fishing.getBukkitEntity() : null; } // Sliver - Player#getFishHook
 }
